extensions/wikia/WikiFactory/WikiFactory.php

```diff
@@ -3408,4 +3408,23 @@ static private function parseValue( $value, $type ) {
 		return var_export( $value, true );
 	}
 
+	static public function getCityLink( $cityId ) {
+		global $wgCityId, $wgSitename;
+
+		$domains = self::getDomains( $cityId );
```

Comment:
For wikis with many domains, the first one is not always (or even usually) the primary domain.  All of them will work but we might as well use the right one.  The `WikiFactory::getHostByDbName` method returns the right one.  We could probably also define a `WikiFactory::getHostById` since we already have a cityId and the first thing `WikiFactory::getHostByDbName` does is convert the dbname into an id.